src/Rules.Framework/Builder/IRuleBuilder.cs

```diff
@@ -24,27 +25,54 @@ public interface IRuleBuilder<TContentType, TConditionType>
         IRuleBuilder<TContentType, TConditionType> WithActive(bool active);
 
         /// <summary>
-        /// Sets the new rule with a specified condition.
+        /// Sets the new rule with the specified condition.
         /// </summary>
         /// <param name="conditionFunc">
         /// The function with specific logic to create the condition for the rule.
         /// </param>
         /// <returns></returns>
-        [Obsolete("This way of adding conditions is being deprecated. Please use a non-deprecated overload.")]
+        [Obsolete("This way of adding conditions is being deprecated. Please use a non-deprecated overload instead.")]
         IRuleBuilder<TContentType, TConditionType> WithCondition(Func<IConditionNodeBuilder<TConditionType>, IConditionNode<TConditionType>> conditionFunc);
 
         /// <summary>
-        /// Sets the new rule with a specified condition.
+        /// Sets the new rule with the specified root condition.
         /// </summary>
         /// <param name="condition">The condition.</param>
         /// <returns></returns>
         IRuleBuilder<TContentType, TConditionType> WithCondition(IConditionNode<TConditionType> condition);
 
+        /// <summary>
+        /// Sets the new rule with a value condition with the specified parameters.
+        /// </summary>
+        /// <typeparam name="TDataType">The type of the data type.</typeparam>
+        /// <param name="conditionType">The content type.</param>
+        /// <param name="condOperator">The operator.</param>
+        /// <param name="operand">The operand.</param>
+        /// <returns></returns>
+        IRuleBuilder<TContentType, TConditionType> WithCondition<TDataType>(TConditionType conditionType, Operators condOperator, TDataType operand);
+
+        /// <summary>
+        /// Sets the new rule with the specified root condition.
+        /// </summary>
+        /// <param name="conditionFunc">The condition func.</param>
+        /// <returns></returns>
+        IRuleBuilder<TContentType, TConditionType> WithCondition(
+            Func<IRootConditionNodeBuilder<TConditionType>, IConditionNode<TConditionType>> conditionFunc);
+
+        /// <summary>
+        /// Sets the new rule with the specified content.
+        /// </summary>
+        /// <param name="contentType">The content type.</param>
+        /// <param name="content">The content.</param>
+        /// <returns></returns>
+        IRuleBuilder<TContentType, TConditionType> WithContent(TContentType contentType, object content);
+
         /// <summary>
         /// Sets the new rule with the specified content container.
         /// </summary>
         /// <param name="contentContainer">The content container.</param>
         /// <returns></returns>
+        [Obsolete("This way of building the content is being deprecated. Please use WithContent().")]
```

Comment:
This one should not be made obsolete, because this is the overload used to satisfy the MongoDB implementation (which uses the `SerializedContentContainer<TContentType>` which inherits from `ContentContainer<TContentType>`).

Check this method for more details:
![image](https://github.com/Farfetch/rules-framework/assets/20577216/5d507bf8-4b68-4b33-bd06-8983951f04ff)

An alternative is to incorporate the extension method `WithSerializedContent<TContentType, TConditionType>(...)` in the `RuleBuilder<TContentType, TConditionType>` itself, and then you can mark this one as obsolete.

Reply:
Followed the suggestion to include the WithSerializedContent in the builder itself, and deleted the extension method.

Do you see any reason to keep the method, since any client will still be able to use it as an extension method without its code breaking? (only way of seeing the user code breaking is if he is using the extension method as a regular static method, which I find it hard to happen).

Reply:
Don't see any reason, I think you can remove it 👌 in fact, this is probably only used by the MongoDB implementation, and will be used most certainly by the SQL Server implementation also.

Thank you for adjusting this!